tools/DatasetExtraction/dataset.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABCMeta, abstractmethod
+from accessify import protected
+from chord import Chord, Song
+import numpy as np
+from glob import glob
+
+class Dataset:
+    #i - song, j - chord
+    songlist = [[]]
+
+
+    @protected
+    def validate(self):
+        pass
+
+    @protected
+    def load(self):
+        pass
+
+    def get_hmm_matrix_i(self):
+        pass
+
+    def get_hmm_matrix_t(self):
+        pass
+
+    def get_hmm_matrix_e(self):
+        pass
+
+
+class Iso_Dataset(Dataset):

```

Comment:
first things first:
1. Each child dataset implementation should have a separate class.
2. Please decide which naming style are you going to use - if it's camel case then you'll have `IsoDataset`.